src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs

```diff
@@ -83,19 +83,45 @@ public async Task<string> DownloadPackageAsync(PackageId packageId,
             bool includePreview = false,
             bool? includeUnlisted = null,
             DirectoryPath? downloadFolder = null,
-            PackageSourceMapping packageSourceMapping = null)
+            PackageSourceMapping packageSourceMapping = null,
+            bool isTool = false)
         {
             CancellationToken cancellationToken = CancellationToken.None;
 
             (var source, var resolvedPackageVersion) = await GetPackageSourceAndVersion(packageId, packageVersion,
                 packageSourceLocation, includePreview, includeUnlisted ?? packageVersion is not null, packageSourceMapping).ConfigureAwait(false);
 
-            FindPackageByIdResource resource = null;
             SourceRepository repository = GetSourceRepository(source);
 
-            resource = await repository.GetResourceAsync<FindPackageByIdResource>(cancellationToken)
-                .ConfigureAwait(false);
+            // TODO: Fix this to use the PackageSearchResourceV3 once https://github.com/NuGet/NuGet.Client/pull/5991 is completed.
+            if (isTool && await repository.GetResourceAsync<ServiceIndexResourceV3>().ConfigureAwait(false) is ServiceIndexResourceV3 serviceIndex)
+            {
+                // See https://learn.microsoft.com/en-us/nuget/api/package-base-address-resource#download-package-manifest-nuspec
+                var uri = serviceIndex.GetServiceEntries("PackageBaseAddress/3.0.0")[0].Uri;
+                var queryUri = uri + $"{packageId}/{packageVersion}/{packageId}.nuspec";
```

Comment:
It looks like you basically went back to the version I initially had, so you might want to glance at baronfel's comment. I do think this should work, though, since my initial version worked.

nits:
```suggestion
                var uri = serviceIndex.GetServiceEntries("PackageBaseAddress/3.5.0")[0].Uri;
                var queryUri = uri + $"{packageId}/{packageVersion}/{packageId}.nuspec?packagetype=dotnettool";
```

If you do it this way, you don't have to do a full check later to see if something is a tool; just check whether it found anything, and you're good.

Reply:
Not quite. The PackageBaseAddress API does not support query strings and the original implementation relied on the length of a specific error message. If the system returned two different error messages with the same content length, that check would not be reliable. Directly checking the nuspec element is more reliable IMO.

Reply:
What is the advantage of using PackageBaseAddress over SearchQueryService? (I hadn't actually noticed that you'd switched to that.)

The length check was my least favorite part of how I did it initially, and I'm definitely glad you removed that. I was just thinking that since you found a way to check whether there's anything in the list of returned objects, that it might be easier to check whether that list is empty rather than looking for dotnettool under one item. This is fine, though.